docs/spec_draft.md

```diff
@@ -485,6 +486,71 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.compare
+
+### Semantics
+
+Performs element-wise comparison of `lhs` and `rhs` tensors according to
+`comparison_direction` and `compare_type`, and produces a `result` tensor. More
+formally, `result[i0, ..., iR-1]` =
+`comparison_direction lhs[i0, ..., iR-1], rhs[i0, ..., iR-1]`.
+
+`comparison_direction` can be one of the following:
+  * `EQ`: `lhs` $=$ `rhs`.
+  * `NE`: `lhs` $\ne$ `rhs`.
+  * `GE`: `lhs` $\ge$ `rhs`.
+  * `GT`: `lhs` $\gt$ `rhs`.
+  * `LE`: `lhs` $\le$ `rhs`.
+  * `LT`: `lhs` $\lt$ `rhs`.
+
+`compare_type` can be one of the following:
+  * `FLOAT`: Standard floating-point comparison of `lhs` and `rhs` as specified
+             in IEEE-754.
+  * `TOTALORDER`: `totalOrder` floating-point comparison of `lhs` and `rhs` as
+                  specified in IEEE-754.
+  * `SIGNED`: Signed comparison of `lhs` and `rhs`.
+  * `UNSIGNED`: Unsigned comparison of `lhs` and `rhs`.
+
+### Inputs
+
+| Name                   | Type                                                    |
+|------------------------|---------------------------------------------------------|
+| `lhs`                  | tensor of any supported type                            |
+| `rhs`                  | tensor of any supported type                            |
+| `comparison_direction` | enum of `EQ`, `NE`, `GE`, `GT`, `LE`, and `LT`          |
+| `compare_type`         | enum of `FLOAT`, `TOTALORDER`, `SIGNED`, and `UNSIGNED` |
+
+### Outputs
+
+| Name     | Type                   |
+|----------|------------------------|
+| `result` | tensor of boolean type |
+
+### Constraints
+
+  * (C1) `lhs` and `rhs` have the same element type.
+  * (C2) `lhs`, `rhs`, and `result` have the same shape.
+  * (C3) Let `E` be the `lhs` element type:
+    * If `E` is complex type, `comparison_direction` $\in$ `{EQ, NE}`,
+      `compare_type` = `FLOAT`.
+    * If `E` is signed integer type, `compare_type` = `SIGNED`.
+    * If `E` is unsigned integer or boolean type, `compare_type` = `UNSIGNED`.
+    * If `E` is floating-type, `compare_type` $\in$ `{FLOAT, TOTALORDER}`.
```

Comment:
From https://github.com/tensorflow/tensorflow/blob/4e0e42544237ce612d5ec3e4dc6a6a8ab8e58df2/tensorflow/compiler/xla/comparison_util.cc#L240 it seems that the TOTALORDER is never being uses:

When I tried the following on the hlo evaluator:
```
func.func @main() -> tensor<i1> {
  %0 = mhlo.constant dense<-0.0> : tensor<f32>
  %1 = mhlo.constant dense<0.0> : tensor<f32>
  %2 = "mhlo.compare"(%0, %1) {compare_type = #mhlo<comparison_type TOTALORDER>, comparison_direction = #mhlo<comparison_direction EQ>}   :(tensor<f32>, tensor<f32>) -> tensor<i1>
  func.return %2 : tensor<i1>
}
```
the result is same as compare_type == FLOAT.

Did you have the same observation about TOTALORDER being unused?

Reply:
XLA uses TOTALORDER, there are examples of comparison that result in different outputs with FLOAT and TOTALORDER

Reply:
From what I can see, [elemental_ir_emitter.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/elemental_ir_emitter.cc) completely ignores `compare_type`?